src/shared-stores/profile-store.tsx

```diff
@@ -188,27 +188,16 @@ export const useProfileStore = create<ProfileStore>()((set, get) => ({
 		}
 
 		const userId = useAuthStore.getState().session?.user.id;
-		const accessToken = useAuthStore.getState().session?.access_token;
 
-		const wateredByUserUrl = `${
-			import.meta.env.VITE_API_ENDPOINT
-		}/get/wateredbyuser?uuid=${userId}`;
-
-		const response = await fetch(wateredByUserUrl, {
-			method: "POST",
-			headers: {
-				Authorization: `Bearer ${accessToken}`,
-				"Content-Type": "application/json",
-			},
-		});
+		const { data, error } = await supabaseClient
+			.rpc("waterings_for_user", { u_id: userId })
+			.select("*");
 
-		if (!response.ok) {
+		if (error) {
```

Comment:
Throwing or use handle?

Reply:
While you are right that this should be more consistent, it was not introduced in this PR. I vote for cleaning this up in a separate PR.